addons/hr_holidays/tests/test_holidays_mail.py

```diff
@@ -66,3 +66,54 @@ def test_email_sent_when_approved(self):
                 admin_emails = self._new_mails.filtered(lambda x: x.partner_ids.employee_ids.id == self.ref('hr.employee_admin'))
                 self.assertEqual(len(admin_emails), 1, "Mitchell Admin should receive an email")
                 self.assertTrue("has been accepted" in admin_emails.preview)
+
+    def test_notify_time_off_officers_on_creation_enabled(self):
+        """When notify_time_off_officers is enabled on the type, time off officers in the same company get notified on creation."""
+        with freeze_time('2022-06-10'):
+            leave_type = self.env['hr.leave.type'].create({
+                'name': 'Paid Time Off',
+                'notify_time_off_officers': True,
+                'requires_allocation': False,
+            })
+
+            with self.mock_mail_gateway():
+                self.env['hr.leave'].create({
+                    'name': 'Paid Time Off Request',
+                    'holiday_status_id': leave_type.id,
+                    'request_date_from': date.today() + relativedelta(days=1),
+                    'request_date_to': date.today() + relativedelta(days=2),
+                    'employee_id': self.employee_emp_id,
+                })
+                self.assertTrue(
+                    any('New Time Off Request' in m.subject for m in self._new_mails),
+                        "Expected officer notification was not sent"
+                    )
+                officer_partner_ids = (self.user_hruser.partner_id | self.user_hrmanager.partner_id).ids
+                recipient_partner_ids = (self._new_mails.mapped('recipient_ids')).ids
```

Comment:
```suggestion
                recipient_partner_ids = self._new_mails.recipient_ids.ids
```
This will also work, no need for `mapped` method